modules/react/modal/stories/examples/FormModal.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import {Modal, useModalModel} from '@workday/canvas-kit-react/modal';
+import {PrimaryButton} from '@workday/canvas-kit-react/button';
+import {HStack} from '@workday/canvas-kit-react/layout';
+import {FormField} from '@workday/canvas-kit-react/form-field';
+import {TextInput} from '@workday/canvas-kit-react/text-input';
+import {plusIcon} from '@workday/canvas-system-icons-web';
+
+export const FormModal = () => {
+  const model = useModalModel();
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault(); // prevent a page reload
+
+    // do form validation here
+
+    console.log('form data', {
+      first: (event.currentTarget.elements.namedItem('first') as HTMLInputElement).value,
+      last: (event.currentTarget.elements.namedItem('last') as HTMLInputElement).value,
+    });
+
+    // if it looks good, submit to the server and close the modal
+    model.events.hide();
+  };
+
+  return (
+    <Modal model={model}>
+      <Modal.Target icon={plusIcon}>Create New User</Modal.Target>
+      <Modal.Overlay>
+        <Modal.Card as="form" onSubmit={onSubmit}>
+          <Modal.CloseIcon aria-label="Close" />
```

Comment:
Do you think `"Close"` still makes sense as the label as opposed to `"Cancel"`?

Reply:
This is currently unspecified. Some form modals may show a "Are you sure you want to close the modal. You will lose unsaved work." due to unsaved work, especially if the form is longer. It depends on the use-case. Some form modals may omit the close icon altogether, and only allow an explicit "Cancel" button.

There are so many examples we could make around form modals. A `shouldHide` model guard could be added to prevent closing if the form isn't submitted (triggering an additional "are you sure model").